core/src/main/scala/akka/kafka/internal/SubSourceLogic.scala

```diff
@@ -278,90 +291,92 @@ private object SubSourceLogic {
 }
 
 /** Internal API */
-private final class SubSourceStage[K, V, Msg](
+@InternalApi
+private abstract class MessageSubSourceLogic[K, V, Msg](
```

Comment:
Is this a `SubSourcLogic` with `MessageBuilder`? Is there a better name?

Reply:
I cleaned this up.

https://github.com/akka/alpakka-kafka/pull/930#discussion_r340823268